types/rational_test.go

```diff
@@ -31,7 +33,10 @@ func TestNewFromDecimal(t *testing.T) {
 		{"1.1", false, NewRat(11, 10)},
 		{"0.75", false, NewRat(3, 4)},
 		{"0.8", false, NewRat(4, 5)},
-		{"0.11111", false, NewRat(11111, 100000)},
+		{"0.11111", false, NewRat(1111, 10000)},
+		{"628240629832763.5738930323617075341", false, NewRat(3141203149163817869, 5000)},
```

Comment:
As I mentioned earlier we should send an error if there are to many decimal places